### PR TITLE
feature: Update PMD to 6.38.0 CY-5026

### PR DESCRIPTION
**Pull request by lolgab**
- Update build to use manual Dockerfile
- Update CircleCI build
- Remove Coverage, scalafix and scapegoat
- Delete redundant Csharp test
- Fix CSharp tests to new tool's output
- Update various dependencies

### CODE REVIEW

File: src/main/scala/com/codacy/duplication/pmd/ScalaTokenizer.scala [hunk 1]
```diff
@@ -15,13 +15,13 @@ private[pmd] object ScalaTokenizer extends Tokenizer {
 
     val fileName = sourceCode.getFileName
 
-    Try(new File(fileName).parse[Source]) match {
-      case Success(Parsed.Success(tree)) =>
+    Try(new File(fileName).parse[Source].toEither) match {
```
Comment (lolgab) [inline src/main/scala/com/codacy/duplication/pmd/ScalaTokenizer.scala:18]: Here I was getting a strange compile error like "The following match could fail on `Success(Success())`"
Converting to `Either` fixes the problem

File: src/main/scala/com/codacy/duplication/pmd/ScalaTokenizer.scala [hunk 2]
```diff
@@ -54,7 +54,7 @@ private[pmd] object ScalaTokenizer extends Tokenizer {
           case _: BOF => "BOF"
           case other  => other.show[Syntax]
         }
-        Seq(new TokenEntry(str, filename, token.pos.start.line + 1))
+        Seq(new TokenEntry(str, filename, token.pos.startLine + 1))
```
Comment (lolgab) [inline src/main/scala/com/codacy/duplication/pmd/ScalaTokenizer.scala:57]: Due to the Scalameta update

File: src/test/scala/com/codacy/duplication/pmd/CpdSpec.scala [hunk 3]
```diff
@@ -120,20 +119,6 @@ class CpdSpec extends Specification {
     }
   }
 
-  private def cSharpTest(): MatchResult[Try[List[DuplicationClone]]] = {
```
Comment (lolgab) [inline src/test/scala/com/codacy/duplication/pmd/CpdSpec.scala:123]: This test was using the same files as the multiple-test so it is redundant.

File: src/main/scala/com/codacy/duplication/pmd/Cpd.scala [hunk 4]
```diff
@@ -135,10 +135,10 @@ object Cpd extends DuplicationTool {
   }
 
   private def duplicationClone(m: Match, rootDirectory: Path): DuplicationClone = {
-    val files: List[DuplicationCloneFile] = m.getMarkSet.asScala.map { mark =>
+    val files: List[DuplicationCloneFile] = m.getMarkSet.asScala.view.map { mark =>
```
Comment (pedrocodacy) [inline src/main/scala/com/codacy/duplication/pmd/Cpd.scala:138]: whats the advantage of having `.view` here? please teach 🙏  🤓

Comment (pedrocodacy) [reply src/main/scala/com/codacy/duplication/pmd/Cpd.scala:138]: is it only a scala 2.13 compatibility thing?

**Review by pedrocodacy — APPROVED** [review]
looks good 👍  

left a small question about a collection operation

File: src/main/scala/com/codacy/duplication/pmd/Cpd.scala (hunk 4 above)
Comment (lolgab) [reply src/main/scala/com/codacy/duplication/pmd/Cpd.scala:138]: Scala `2.13` replaced `collection.breakOut` with views: https://docs.scala-lang.org/overviews/core/collections-migration-213.html

Comment (pedrocodacy) [reply src/main/scala/com/codacy/duplication/pmd/Cpd.scala:138]: 🤘

**Pull request merged**